File: src/main/java/hse/btf/pdfeditor/models/itemsjava/TextItem.java
package hse.btf.pdfeditor.models.itemsjava;

import hse.btf.pdfeditor.models.Item;
import javafx.beans.property.SimpleStringProperty;
import javafx.beans.property.StringProperty;
import javafx.beans.value.ObservableStringValue;

public class TextItem extends Item {
    public StringProperty text = new SimpleStringProperty("type text here");
}

File: src/main/kotlin/hse/btf/pdfeditor/models/itemsstand/TextItem.kt
package hse.btf.pdfeditor.models.itemsstand

import hse.btf.pdfeditor.models.serializers.IntegerPropertySerializer
import hse.btf.pdfeditor.models.serializers.StringPropertySerializer
import javafx.beans.property.IntegerProperty
import javafx.beans.property.SimpleIntegerProperty
import javafx.beans.property.SimpleStringProperty
import javafx.beans.property.StringProperty
import kotlinx.serialization.Serializable

@Serializable
data class TextItem(
    @Serializable(with = StringPropertySerializer::class)
    val text: StringProperty = SimpleStringProperty("type text here"),

    @Serializable(with = IntegerPropertySerializer::class)
    val fontSize: IntegerProperty = SimpleIntegerProperty(13)
) : Item()